### PR TITLE
config/jobs: move cluster-api-provider-azure to main branch

### PR DESCRIPTION
**Pull request by mboersma**
Refs kubernetes-sigs/cluster-api-provider-azure#749

### CODE REVIEW

**Comment by mboersma** [comment]
/hold

This needs to be merged in conjunction with the actual branch renaming.

**Review by CecileRobertMichon — COMMENTED** [review]
/ok-to-test

File: config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml [hunk 1]
```diff
@@ -261,7 +261,7 @@ presubmits:
     extra_refs:
     - org: jackfrancis #TODO change back to kubernetes-sigs
       repo: cluster-api-provider-azure
-      base_ref: capz-ha-control-plane-tests #TODO change back to master
+      base_ref: capz-ha-control-plane-tests #TODO change back to main
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:264]: @jackfrancis do you still need the fork here?

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve
/hold until tomorrow when we do the rename

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml [hunk 2]
```diff
@@ -249,7 +249,7 @@ presubmits:
     extra_refs:
     - org: jsturtevant # TODO(jstur): change it to 'kubernetes-sigs'
       repo: cluster-api-provider-azure
-      base_ref: windows-custom-binaries # TODO(jstur): change it to 'master'
+      base_ref: windows-custom-binaries # TODO(jstur): change it to 'main'
```
Comment (jsturtevant) [inline config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml:252]:
```suggestion
      base_ref: main # TODO(jstur): change it to 'main'
```

**Review by jsturtevant — CHANGES_REQUESTED** [review]

**Comment by jsturtevant** [comment]
Can we also include the Windows jobs.  They merged today so might have been missed

 https://github.com/kubernetes/test-infra/blob/70d200dac3a6f2efd119e38bff0aae0963df01b5/config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml#L237 

https://github.com/kubernetes/test-infra/blob/e13b6b7d90109de50b8d50ede82e4674c03b22e6/config/jobs/kubernetes-sigs/sig-windows/release-1.22-windows.yaml#L57

File: config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml (hunk 2 above)
Comment (mboersma) [reply config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml:252]: Should I revert the org to `kubernetes-sigs` as well? I'm assuming we want off the fork entirely.

Comment (jsturtevant) [reply config/jobs/kubernetes-sigs/sig-windows/release-master-windows-presubmits.yaml:252]: yes please!  It merged into capz so we are ready to cut over

**Comment by mboersma** [comment]
Updated for new Windows jobs, thanks for the eagle eyes @jsturtevant.

File: config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml (hunk 1 above)
Comment (CecileRobertMichon) [reply config/jobs/kubernetes/sig-cloud-provider/azure/release-master.yaml:264]: will follow up on this later

**Comment by CecileRobertMichon** [comment]
/assign @jsturtevant for sig-windows jobs
/assign @dims @spiffxp (not sure how to assign on-call?) for image pushing job approval


/lgtm 
for cloud-provider and capz jobs

**Comment by CecileRobertMichon** [comment]
@jsturtevant we need your approve too for windows files

**Pull request merged**